### ISSUE TITLE
Support for Libre 2

### ISSUE THREAD

**Issue by DreadRoberts** [tidepool-org/chrome-uploader#903]
I have a libre 2 device and would like to upload my data in Tidepool. Do you have any plans to support the Libre 2?

**Comment by gniezen** [comment]
We would love to support the Libre 2, but after some initial investigation it seems Abbott has made significant changes to the device protocol, so this will not be easy to implement. I hope that we'll get there eventually, but don't have any estimate on availability yet.

**Comment by DreadRoberts** [comment]
Is there any way I could help?

**Comment by gniezen** [comment]
Do you have experience with decoding packets using Wireshark?

**Comment by DreadRoberts** [comment]
Sorry for the late response! I am not very experienced but had a look at the communication and it looks like abbott is using the serial number of the device to do some sort of encryption of the packages

**Comment by gniezen** [comment]
@DreadRoberts Yeah, I looked into this myself last month, and while the Libre 2 reader appears to use the same HID protocol and initialization sequence as the Libre 1 reader, all communications after the initialization sequence appears to be encrypted. What makes you think they're using the serial number as an encryption key?

**Comment by DreadRoberts** [comment]
The last and only message which is transmited unencrypted is the serial number.

**Comment by DreadRoberts** [comment]
I commented on another project what my current findings are, if you are interested:
https://github.com/glucometers-tech/glucometer-protocols/issues/8

**Comment by tolot27** [comment]
I'll have a look on it.

**Comment by MPM1107** [comment]
I've recently written an implementation of the encrypted protocol for a python library called freestyle-hid. I found this repo through that one because this issue was linked in one of the issues from that library that was also about the libre 2. While I don't have the time to get into this codebase, I'd be happy to provide some guidance if someone else wants to implement support.
